test/powershell/Modules/Microsoft.PowerShell.Utility/Add-Type.Tests.ps1

```diff
@@ -26,41 +26,212 @@ Describe "Add-Type" -Tags "CI" {
+    It "Should not throw given a simple C# class definition" {
+        # Also we check that '-Language CSharp' is by default.
+        # In subsequent launches from the same session
+        # the test will be passed without real compile - it will return an assembly previously compiled.
+        { Add-Type -TypeDefinition "public static class CSharpfooType { }" } | Should Not Throw
+        [CSharpfooType].Name | Should BeExactly "CSharpfooType"
+    }
+
+    It "Should not throw given a simple VisualBasic class definition" {
+        # In subsequent launches from the same session
+        # the test will be passed without real compile - it will return an assembly previously compiled.
+        { Add-Type -TypeDefinition "Public Class VBfooType `n End Class" -Language VisualBasic } | Should Not Throw
+        [VBfooType].Name | Should BeExactly "VBfooType"
     }
 
     It "Can use System.Management.Automation.CmdletAttribute" {
         $code = @"
-[System.Management.Automation.Cmdlet("Get", "Thing", ConfirmImpact = System.Management.Automation.ConfirmImpact.High, SupportsPaging = true)]
-public class AttributeTest$guid {}
+using System.Management.Automation;
+[System.Management.Automation.Cmdlet("Get", "Thing$guid", ConfirmImpact = System.Management.Automation.ConfirmImpact.High, SupportsPaging = true)]
+public class AttributeTest$guid : PSCmdlet
+{
+    protected override void EndProcessing()
+
+    {
+        WriteObject("$guid");
+    }
+}
 "@
-        Add-Type -TypeDefinition $code -PassThru | Should Not Be $null
+        $cls = Add-Type -TypeDefinition $code -PassThru | Select-Object -First 1
+        $testModule = Import-Module $cls.Assembly -PassThru
+
+        Invoke-Expression -Command "Get-Thing$guid" | Should BeExactly $guid
+
+        Remove-Module $testModule -ErrorAction SilentlyContinue -Force
     }
 
     It "Can load TPA assembly System.Runtime.Serialization.Primitives.dll" {
-        Add-Type -AssemblyName 'System.Runtime.Serialization.Primitives' -PassThru | Should Not Be $null
+        $returnedTypes = Add-Type -AssemblyName 'System.Runtime.Serialization.Primitives' -PassThru
+        $returnedTypes.Count | Should BeGreaterThan 0
+        ($returnedTypes[0].Assembly.FullName -Split ",")[0]  | Should BeExactly 'System.Runtime.Serialization.Primitives'
+    }
+
+    It "Can compile <sourceLanguage> files" -TestCases @(
+        @{
+            type1 = "[Test.AddType.CSharpTest1$guid]"
+            type2 = "[Test.AddType.CSharpTest2$guid]"
+            file1 = $CSharpFile1
+            file2 = $CSharpFile2
+            sourceLanguage = "CSharp"
+        }
+        @{
+            type1 = "[Test.AddType.VBTest1$guid]"
+            type2 = "[Test.AddType.VBTest2$guid]"
+            file1 = $VBFile1
+            file2 = $VBFile2
+            sourceLanguage = "VisualBasic"
+        }
+    ) {
+        param($type1, $type2, $file1, $file2, $sourceLanguage)
+
+        { $type1 = Invoke-Expression -Command $type1 } | Should Throw
```

Comment:
this could be done also as:
```
$type1 -as [type] | Should BeNullOrEmpty
```
etc, yes?

Reply:
Good catch! Thanks!

Fixed.